recipe/conda_build_config.yaml

```diff
@@ -1,7 +1,7 @@
 # https://conda-forge.org/docs/maintainer/knowledge_base/#requiring-newer-macos-sdks
 # https://conda-forge.org/news/2024/03/24/stdlib-migration/
-MACOSX_SDK_VERSION:  # [osx and x86_64]
-  - 11.0            # [osx and x86_64]
+MACOSX_SDK_VERSION:            # [osx and x86_64]
+  - 13.3                       # [osx and x86_64]
 c_stdlib_version:              # [osx and x86_64]
   - 11.0                       # [osx and x86_64]
```

Comment:
I was concerned that these didn't match, but I think this is correct. From the [conda-forge docs](https://conda-forge.org/docs/maintainer/knowledge_base/#requiring-newer-macos-sdks), we don't need to bump `c_stdlib_version` as long as we're not getting additional errors.